### ISSUE TITLE
bug: update APISIX helm chart from 0.11.4 to 0.12.3 fails

### ISSUE THREAD

**Issue by mkyc** [apache/apisix-ingress-controller#1575]
### Issue description

After upgrade [apisix helm chart](https://artifacthub.io/packages/helm/apisix/apisix) from version `0.11.4` to version `0.12.3` I get following errors in `ingress-controller` pod: 

```text
2022-12-30T22:40:19+08:00    error    apisix/ssl.go:120    failed to list ssl: unexpected status code 404; error message: {"error_msg":"not found"}

2022-12-30T22:40:19+08:00    error    apisix/cluster.go:232    failed to list ssl in APISIX: unexpected status code 404; error message: {"error_msg":"not found"}

2022-12-30T22:40:19+08:00    error    apisix/cluster.go:182    failed to sync cache    {"cost_time": "8.092071269s", "cluster": "default"}
2022-12-30T22:40:19+08:00    error    providers/controller.go:387    failed to wait the default cluster to be ready: unexpected status code 404; error message: {"error_msg":"not found"}

2022-12-30T22:40:19+08:00    info    apisix/cluster.go:172    syncing cache    {"cluster": "default"}
2022-12-30T22:40:19+08:00    error    apisix/route.go:118    failed to list routes: Get "http://apisix-admin.apisix.svc.cluster.local:9180/apisix/admin/routes": context canceled
2022-12-30T22:40:19+08:00    error    apisix/cluster.go:222    failed to list routes in APISIX: Get "http://apisix-admin.apisix.svc.cluster.local:9180/apisix/admin/routes": context canceled
2022-12-30T22:40:19+08:00    error    apisix/cluster.go:182    failed to sync cache    {"cost_time": "84.161µs", "cluster": "default"}
2022-12-30T22:40:19+08:00    info    apisix/cluster.go:384    syncing schema    {"cluster": "default"}
2022-12-30T22:40:19+08:00    error    apisix/plugin.go:46    failed to list plugins' names: Get "http://apisix-admin.apisix.svc.cluster.local:9180/apisix/admin/plugins?all=true": context canceled
2022-12-30T22:40:19+08:00    error    apisix/cluster.go:404    failed to list plugin names in APISIX: Get "http://apisix-admin.apisix.svc.cluster.local:9180/apisix/admin/plugins?all=true": context canceled
2022-12-30T22:40:19+08:00    error    apisix/cluster.go:367    failed to sync schema: Get "http://apisix-admin.apisix.svc.cluster.local:9180/apisix/admin/plugins?all=true": context canceled
2022-12-30T22:40:19+08:00    info    providers/controller.go:189    controller now is running as a candidate    {"namespace": "apisix", "pod": "apisix-ingress-controller-75fc6dd-52td8"}
I1230 22:40:19.853714       1 leaderelection.go:248] attempting to acquire leader lease apisix/ingress-apisix-leader...
2022-12-30T22:40:19+08:00    info    providers/controller.go:137    LeaderElection    {"message": "apisix-ingress-controller-75fc6dd-52td8 became leader", "event_type": "Normal"}
I1230 22:40:19.865350       1 leaderelection.go:258] successfully acquired lease apisix/ingress-apisix-leader
2022-12-30T22:40:19+08:00    info    providers/controller.go:359    controller tries to leading ...    {"namespace": "apisix", "pod": "apisix-ingress-controller-75fc6dd-52td8"}
2022-12-30T22:40:19+08:00    warn    apisix/cluster.go:344    waiting cluster default to ready, it may takes a while
2022-12-30T22:40:19+08:00    info    apisix/cluster.go:384    syncing schema    {"cluster": "default"}
2022-12-30T22:40:19+08:00    info    apisix/cluster.go:172    syncing cache    {"cluster": "default"}
2022-12-30T22:40:19+08:00    error    apisix/ssl.go:120    failed to list ssl: unexpected status code 404; error message: {"error_msg":"not found"}

2022-12-30T22:40:19+08:00    error    apisix/cluster.go:232    failed to list ssl in APISIX: unexpected status code 404; error message: {"error_msg":"not found"}


```

### Environment

```shell
$ kubectl get pod -n apisix -o jsonpath="{.items[*].spec.containers[*].image}" | tr -s '[[:space:]]' '\n' | sort | uniq -c
   1 apache/apisix-dashboard:2.13-alpine
   1 apache/apisix-ingress-controller:1.6.0
   1 apache/apisix:3.1.0-debian
   3 docker.io/bitnami/etcd:3.5.4-debian-11-r14
```

### Minimal test code / Steps to reproduce

`values.yaml`: 

```yaml
dashboard:
  enabled: true
  service:
    type: NodePort

ingress-controller:
  enabled: true
  config:
    apisix:
      serviceNamespace: apisix

apisix:
  httpRouter: radixtree_host_uri

configurationSnippet:
  httpSrv: |
    set $session_secret XXX;

```

### Actual result

n/a

### Error log

already provided

**Comment by Randoooom** [comment]
Can confirm this issue. Seems like the admin-api uses v3 as default version now, while the ingress-controller chart still has v2 as default api version (https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix-ingress-controller/values.yaml#L134). Overwriting this default value to v3 resolved that specific issue for me:
```yaml
ingress-controller:
  config:
    apisix:
      adminAPIVersion: v3
```

I still have a CrashLoop using this configuration, but this seems to be a different error.

**Comment by mkyc** [comment]
In my case I do not have CrashLoop. `ingress-controller` pod is just producing this log every 2 seconds but it doesn't crash. I'm checking your hint right now.

**Comment by mkyc** [comment]
Hmmmm .... interestingly with additional configuration you provided it works on my side right now.

**Comment by tokers** [comment]
@tao12345666333 I remember apisix ingress controller 1.6 supports apisix 3.x? Right?

**Comment by tao12345666333** [comment]
yes. 

If you want to upgrade APISIX 3.0+ . You should add adminAPIVersion=v3 configuration

**Comment by mkyc** [comment]
since version `0.12.3` helm chart is using APISIX versions 3.x, shouldn't this value `adminAPIVersion: v3` be default?

**Comment by tao12345666333** [comment]
yes!! I want to split out the helm chart for APISIX v2 and v3 https://github.com/apache/apisix-helm-chart/issues/419

**Comment by tao12345666333** [comment]
I have released new 1.1.0 version helm chart. Please run `helm repo update` to fetch new chart.

It includes APISIX v3 and APISIX Ingress v1.6.

Or you can using 0.11.5, it includes APISIX v2 and APISIX Ingress v1.6.  @mkyc

**Comment by tao12345666333** [comment]
I have removed all 0.12.x version. ref: https://github.com/apache/apisix-helm-chart/issues/448

Let's close this one.  Pls run `helm repo update` to fetch new version helm charts. Thanks
